Sources/Redis/Redis+Cache.swift

```diff
@@ -0,0 +1,45 @@
+import Vapor
+
+extension Application.Caches {
+    public var redis: Cache {
+        RedisCache(client: self.application.redis)
+    }
+}
+
+extension Application.Caches.Provider {
+    public static var redis: Self {
+        .init {
+            $0.caches.use { $0.caches.redis }
+        }
+    }
+}
+
+private struct RedisCache: Cache {
```

Comment:
Why not generic to avoid existential?

Reply:
Hmm... can you elaborate on this a bit more?

Reply:
I was suggesting that instead of storing the client as an existential `RedisClient`, why not make the `RedisCache` generic?

```swift
struct RedisCache<Client: RedisClient>: Cache {
```
and likewise use `var redis: some Cache`

as it avoids all of the dynamic dispatch of existentials